plugin/trino-memsql/src/main/java/io/trino/plugin/memsql/MemSqlClient.java

```diff
@@ -624,17 +626,25 @@ protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCon
         }
 
         String typeName = typeHandle.getJdbcTypeName().get();
-        if (typeName.equalsIgnoreCase("tinyint unsigned")) {
-            return Optional.of(smallintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("smallint unsigned")) {
-            return Optional.of(integerColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("int unsigned")) {
-            return Optional.of(bigintColumnMapping());
-        }
-        if (typeName.equalsIgnoreCase("bigint unsigned")) {
-            return Optional.of(decimalColumnMapping(createDecimalType(20)));
+        if (UNSIGNED_TYPE_REGEX.matcher(typeName).matches()) {
+            switch (typeHandle.getJdbcType()) {
+                case Types.BIT:
+                    return Optional.of(booleanColumnMapping());
+
+                case Types.TINYINT:
+                    return Optional.of(smallintColumnMapping());
+
+                case Types.SMALLINT:
+                    return Optional.of(integerColumnMapping());
+
+                case Types.INTEGER:
+                    return Optional.of(bigintColumnMapping());
+
+                case Types.BIGINT:
+                    return Optional.of(decimalColumnMapping(createDecimalType(20)));
+                default:
+                    throw new IllegalStateException("Cannot find mapping for " + typeHandle);
```

Comment:
Q: Should I remove this?
Original version just fell through if nothing matched, not sure if this is safer to fail fast?

Reply:
IMO remove, looking at other connector implementations `Optional.empty()` is used to represent lack of support for a data type. 
Example for postgres: https://github.com/trinodb/trino/blob/master/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java#L543

Reply:
The idea is to not make the table inaccessible just because of a single unsupported type. Other columns are queryable so in Trino the unsupported column would appear missing. Also see the configs under https://trino.io/docs/current/connector/postgresql.html#id2 for how Trino can try to map/read those rows.

Reply:
Makes sense - thanks! 👍